### ISSUE TITLE
[DinoMod] Egg recipes cannot use dino eggs

### ISSUE THREAD

**Issue by wapcaplet** [CleverRaven/Cataclysm-DDA#46365]
# Describe the bug

When crafting recipes with an egg requirement, dino eggs cannot be used. Even simple recipes like boiled, scrambled, or fried eggs cannot be made from dino eggs.

# Steps To Reproduce

1. Get some dino eggs
2. Be unable to cook them before eating

# Expected behavior

Dino eggs should be interchangeable with other reptile or bird eggs.

# Screenshots

Here I have some eggs from edmontonia and torosaurus:

![image](https://user-images.githubusercontent.com/238395/103179361-7738fa00-4848-11eb-82a6-0221c2932677.png)

![image](https://user-images.githubusercontent.com/238395/103179366-90da4180-4848-11eb-9bd4-971c9209665f.png)

But the eggs are not listed as eligible ingredients, so I cannot cook them:

![image](https://user-images.githubusercontent.com/238395/103179379-af403d00-4848-11eb-855d-18eba680c9f1.png)

# Versions and configuration

- OS: Linux
    - OS Version: Distributor ID: Pop; Description: Pop!_OS 20.04 LTS; Release: 20.04; Codename: focal; 
- Game Version: 0.E-8509-g35c0af53e0 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Bionic Professions [package_bionic_professions],
    DinoMod [DinoMod],
    No Rail Stations [No_Rail_Stations],
    Generic Guns [generic_guns],
    Magiclysm [magiclysm]
]

# Additional context

Previously reported as issue #42054 and fixed by PR #42583

**Comment by wapcaplet** [comment]
Possibly this is because I also have Magiclysm loaded? My list of eggs appears to include *only* the Magiclysm eggs (including owlbear).

**Comment by wapcaplet** [comment]
When using my attached savegame as a template for a new world, the recipe issue persists - dino eggs are not included in recipe components.

However, when starting a new world with the same mods, I cannot reproduce the problem - recipes *do* include dino eggs.

**Comment by LyleSY** [comment]
Yes, this is a limitation for how ingredients lists are currently managed by vanilla. I assumed the list was modding friendly and would allow mods to add to the list, but actually it requires writing over the list, with each mod that touches the list fighting for list supremacy. Harrumph that Magiclysm wins because I'm pretty sure DinoMod innovated this, but the fix as I understand it is to not play both at the same time (boo) or fix how these lists are managed by the main game to be more friendly for egg cooking modding (yay)

**Comment by LyleSY** [comment]
Oh funny, just read your new comment. Maybe this was already fixed?

**Comment by Maleclypse** [comment]
I tried to look at making requirements able to add opposed to override and it was way beyond my skill level. For Aftershock and Magiclysm I make a patch mod on my repo to get past this limitation.

**Comment by LyleSY** [comment]
Just confirmed this is still an issue at this time, Magiclysm owl bear eggs are the only modded egg ingredients appearing when both mods are activated, specifically this file https://github.com/CleverRaven/Cataclysm-DDA/blob/master/data/mods/Magiclysm/cooking_components.json
